### PR TITLE
feat: add image get and image list

### PR DESCRIPTION
**Pull request by kemingy**
Signed-off-by: Keming <kemingyang@tensorchord.ai>

* to support https://github.com/tensorchord/envd/issues/1027#issuecomment-1290226640

### CODE REVIEW

File: api/types/image.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package types
+
+import (
+	dockertypes "github.com/docker/docker/api/types"
```
Comment (gaocegege) [inline api/types/image.go:8]: I am not sure if we should use docker types.ImageSummary definition, since our API is not docker specific.

Comment (kemingy) [reply api/types/image.go:8]: I see. BTW, the `ImageClient` interface is using `types.ImageSummary`. Should we refactor these?

Comment (gaocegege) [reply api/types/image.go:8]: I think so. We should refactor the ImageClient.

Comment (kemingy) [reply api/types/image.go]: Updated. PTAL.

**Review by gaocegege — APPROVED** [review]

**Pull request merged**